.github/workflows/publishDockerToGhcr.yml

```diff
@@ -0,0 +1,41 @@
+name: Create and publish a Docker image
+
+on:
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
```

Comment:
Can we use tagged releases here?

Reply:
We can.  Github recommends [against using tags](https://securitylab.github.com/research/github-actions-building-blocks/#:~:text=By%20a%20tag/release%3A%20uses%3A%20owner/action%2Dname%40v1.%20This%20option%20protects%20you%20from%20unintentional%20changes%2C%20however%2C%20it%20is%20prone%20to%20intentional%20modifications.%20The%20tag%20can%20be%20changed%20later%20to%20point%20to%20a%20different%20changeset%20if%2C%20for%20example%2C%20the%20creator%20of%20the%20action%20is%20compromised%20or%20the%20maintainer%20changes.) because they could be abused by someone in a pipeline injection attack.  I updated the file with a comment on each line containing the tag that the commit hash references.  If we want to trust these actions, and I think that's sensible in this scenario given how frequently these are used in github, we can now quickly switch to the tag.

Reply:
Seems a bit paranoid, but definitely justifiable. Thanks for explanation! The obvious disadvantage though is not getting (also security) updates automatically...

 I'll consider adopting it to my projects and repositories.